src/layout/TagsInput.js

```diff
@@ -8,9 +8,19 @@ import { Flex } from './Flex'
 import Popover from './Popover'
 import OutsideClickHandler from 'react-outside-click-handler'
 
-let isValidTag = (tag, tags) => {
-  let cleanTag = _.trim(tag)
-  return !_.isEmpty(cleanTag) && !_.includes(cleanTag, tags)
+let isValidInput = (tag, tags) => !_.isEmpty(tag) && !_.includes(tag, tags)
+
+let getUniqueTagsFromInput = (tag, tags, splitCommas = true) => {
+  let trimmedTag = _.trim(tag)
+  if (splitCommas && _.includes(',', trimmedTag)) {
+    // Extract only the unique and not present in the current tags collection tags from the string
+    let newUniqTags = _.difference(
+      _.uniq(_.compact(_.split(',', trimmedTag))),
+      tags
+    )
+    return _.isEmpty(newUniqTags) ? null : newUniqTags.join(',')
```

Comment:
This is a little weird at least from a naming perspective - we're saying that if `splitCommas` is true, we're still returning a string joined by a comma 🤔

Reply:
Added comments and renamed the function.

Reply:
I think the issue is that, in the case of `splitCommas`, we're taking a string, splitting it, deduping it, joining it, splitting it, iterating, and then adding. Let's just improve the logic on line 102 where that's happening so we don't do extra work